src/workers/postUpdated.ts

```diff
@@ -294,19 +301,27 @@ const updatePost = async ({
   }
 
   const title = data?.title || databasePost.title;
-  const updateBecameVisible =
-    content_type === PostType.Freeform
-      ? databasePost.visible
-      : !databasePost.visible && !!title?.length;
 
   data.id = databasePost.id;
   data.title = title;
-  data.visible = updateBecameVisible;
-  data.visibleAt = updateBecameVisible
-    ? databasePost.visibleAt ?? data.metadataChangedAt
-    : null;
   data.sourceId = data.sourceId || databasePost.sourceId;
 
+  let updateBecameVisible = false;
+
+  if (!databasePost.visible) {
+    updateBecameVisible = getPostVisible({ post: { title } });
+  }
+
+  if (updateBecameVisible) {
+    data.visible = true;
+  } else {
+    data.visible = databasePost.visible;
+  }
```

Comment:
Post can become visible only once, In all other cases (if already visible or still `false` we just use the current values from DB.